src/components/Charts/ChartContainer.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { localPoint } from '@vx/event';
+import { Group } from '@vx/group';
+import { useTooltip } from '@vx/tooltip';
+import HoverOverlay from './HoverOverlay';
+import * as Style from './Charts.style';
+
+const ChartContainer = <T extends unknown>({
+  width,
+  height,
+  marginTop = 5,
+  marginBottom = 40,
+  marginLeft = 40,
+  marginRight = 5,
+  x,
+  y,
+  data,
+  renderMarker,
+  renderTooltip,
+  children,
+}: {
+  width: number;
+  height: number;
+  marginTop?: number;
+  marginBottom?: number;
+  marginLeft?: number;
+  marginRight?: number;
+  x: (d: T) => number;
+  y: (d: T) => number;
+  data: T[];
+  renderMarker: any;
+  renderTooltip: any;
+  children: React.ReactNode;
+}) => {
+  const chartWidth = width - marginLeft - marginRight;
+  const chartHeight = height - marginTop - marginBottom;
+
+  const tooltip = useTooltip<T>();
```

Comment:
what's the benefit of the <T> here? guess i'm a lil confused

Reply:
`useTooltip` is a hook with an internal state, it stores the currently hovered data point. Since this logic is used for more than one type of chart, the data of the hovered point might have a different type. For the R(t) chart, the shape is `PointRt = { x: date, y: { low: number, high: number, rt: number }}`, for the other charts is `Point = { x: date, y: number }`

If we don't use the generic, TS is going to complain because we later try to access the type-specific properties when rendering the tooltip